src/powershell/Public/Remove-FinOpsHub.ps1

```diff
@@ -0,0 +1,106 @@
+
+    .PARAMETER InputObject
+    Required when specifying InputObject. Expected object is the output of Get-FinOpsHub.
+
+    .PARAMETER KeepStorageAccount
+    Optional. Indicates that the storage account associated with the FinOps Hub should be retained.
+
+    .EXAMPLE
+    Remove-FinOpsHub -Name MyHub -ResourceGroupName MyRG -KeepStorageAccount
+
+    Deletes a FinOps Hub named MyHub and deletes all associated resource except the storagea ccount.
+#>
+
+function Remove-FinOpsHub
+{
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param
+    (
+        [Parameter(Mandatory = $true, ParameterSetName = 'Name')]
+        [ValidateNotNullOrEmpty ()]
+        [string]
+        $Name,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [string]
+        $ResourceGroupName,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'Object')]
+        [ValidateNotNullOrEmpty ()]
+        [psobject]
+        $InputObject,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $KeepStorageAccount,
+
+        [Parameter(ParameterSetName = 'Name')]
+        [Parameter(ParameterSetName = 'Object')]
+        [switch]
+        $Force
+    )
+
+    $context = Get-AzContext
+    if (-not $context)
+    {
+        throw $script:localizedData.ContextNotFound
+    }
+
+    try
+    {
+
+        if ($PSCmdlet.ParameterSetName -eq 'Name')
+        {
+            if (-not [string]::IsNullOrEmpty($ResourceGroupName))
+            {
+                $hub = Get-FinOpsHub -Name $Name -ResourceGroupName $ResourceGroupName
+            }
+            else
+            {
+                $hub = Get-FinOpsHub -Name $Name
+                $ResourceGroupName = $hub.Resources[0].ResourceGroupName
+            }
+        }
+        else
+        {
+            $hub = $InputObject
+            $Name = $hub.Name
+            $ResourceGroupName = $hub.Resources[0].ResourceGroupName
+        }
+
+        if (-not $hub)
+        {
+            throw $script:localizedData.FinOpsHubNotFound -f $Name
+        }
+
+        # Extract unique identifier from Key Vault name - In the future to use Get-FinOpsHub when all resources are returned.
+        $kv = ($hub.Resources | Where-Object ResourceType -eq "Microsoft.KeyVault/vaults")[0].Name
+        $uniqueId = $kv.Substring($kv.LastIndexOf('-') + 1)
+        Write-Verbose -Message "Unique identifier: $uniqueId"
+
+        $resources = Get-AzResource -ResourceGroupName $ResourceGroupName | Where-Object Name -like "*$uniqueId*" | Where-Object {(-not $KeepStorageAccount) -or $_.ResourceType -ne "Microsoft.Storage/storageAccounts"}
+
+        if ($PSCmdlet.ShouldProcess($Name, 'DeleteFinOpsHub'))
+        {
+            $resources | Remove-AzResource -Force:$Force
+        }
+    }
+    catch
+    {
+        Write-Verbose -Message $($LocalizedData.ErrorRegisteringProvider -f $_.Exception.Message)
```

Comment:
Verbose should only be used for verbose messages. It's being used here to output the originating exception. I would remove this and add the exception to the throw statement below it.